test/OpenTelemetry.Extensions.Docker.Tests/Resources/DockerResourceDetectorTests.cs

```diff
@@ -85,6 +92,12 @@ public void TestValidContainer()
             tempFile.Write(CGROUPLINE);
             Assert.Equal(CONTAINERID, this.GetContainerId(dockerResourceDetector.BuildResource(tempFile.FilePath)));
         }
+
+        using (TempFile tempFile = new TempFile())
+        {
+            tempFile.Write(CGROUPLINEV2);
+            Assert.Equal(CONTAINERIDV2, this.GetContainerId(dockerResourceDetector.BuildResource(tempFile.FilePath, false)));
```

Comment:
I'd use Detect() method for unit test instead of BuildResource(). Detect is part of the interface while BuildResource is just a part of internal implementation

Reply:
Right now there is currently no reference to Detect in the tests, and I believe its due to considerations of writing to a specific file path on the machine. As such, the only aspect of the test that would be lost by not including this detect is the reading of the file at the designated filepath. I'm not sure if its a good idea to include Detect in the tests if this is the case.

Reply:
Yes you are right. If we don't want to stick to the FILEPATH and FILEPATH2 locations you cannot use Detect() method. In that case I think you made the right choice to test BuildResourceV1 and BuildResourceV2 separately. At the end if you know which version you are writing to a temp file you should know which method to call to read from the file